src/util/cmdlineargs.cpp

```diff
@@ -36,92 +38,198 @@ CmdlineArgs::CmdlineArgs()
 }
 
 namespace {
-    bool parseLogLevel(
-        QLatin1String logLevel,
-        mixxx::LogLevel *pLogLevel) {
-        if (logLevel == QLatin1String("trace")) {
-            *pLogLevel = mixxx::LogLevel::Trace;
-        } else if (logLevel == QLatin1String("debug")) {
-            *pLogLevel = mixxx::LogLevel::Debug;
-        } else if (logLevel == QLatin1String("info")) {
-            *pLogLevel = mixxx::LogLevel::Info;
-        } else if (logLevel == QLatin1String("warning")) {
-            *pLogLevel = mixxx::LogLevel::Warning;
-        } else if (logLevel == QLatin1String("critical")) {
-            *pLogLevel = mixxx::LogLevel::Critical;
-        } else {
-            return false;
-        }
-        return true;
+bool parseLogLevel(
+        const QString& logLevel,
+        mixxx::LogLevel* pLogLevel) {
+    if (logLevel.compare(QLatin1String("trace"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Trace;
+    } else if (logLevel.compare(QLatin1String("debug"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Debug;
+    } else if (logLevel.compare(QLatin1String("info"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Info;
+    } else if (logLevel.compare(QLatin1String("warning"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Warning;
+    } else if (logLevel.compare(QLatin1String("critical"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Critical;
+    } else {
+        return false;
     }
+    return true;
 }
+} // namespace
+
+bool CmdlineArgs::Parse(const QStringList& arguments) {
+    QCommandLineParser parser;
+    parser.setApplicationDescription(QCoreApplication::translate("main",
+            "Mixxx is an open source DJ software. For more information, see "
+            "https://manual.mixxx.org/2.3/chapters/appendix.html#command-line-options)"));
+    parser.setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
+
+    // add options
+    const QCommandLineOption fullscreen(QStringList() << "f"
+                                                      << "fullScreen",
+            QCoreApplication::translate("main", "Starts Mixxx in full-screen mode"));
+    parser.addOption(fullscreen);
+
+    const QCommandLineOption locale(QStringLiteral("locale"),
+            QCoreApplication::translate("main",
+                    "Use a custom locale for loading translations. (e.g "
+                    "'fr')"));
+    parser.addOption(locale);
+
+    // An option with a value
+    const QCommandLineOption settingsPath(QStringLiteral("settingsPath"),
```

Comment:
I'd love to get rid of these weird camelCase command line args and use names like `--fullscreen` and `--settings-path` instead. However, out of scope for this PR.

```suggestion
    const QCommandLineOption settingsPath(QStringLiteral("settingsPath"),
```

Reply:
I added non camecase versions and added a comment that CamelCase arguments will be removed in 2.5

Reply:
I'm okay with removing the camelCase versions for 2.4. I don't think there's any particular reason to delay that another release. What do others think?

Reply:
Let's clean it up now.

Reply:
Yes, I think it would be likely to be forgotten or procrastinated on finishing if we don't do it now.